apps/web/pages/api/auth/signup.ts

```diff
@@ -116,12 +117,18 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         });
       }
 
-      const membership = await prisma.membership.update({
+      const membership = await prisma.membership.upsert({
         where: {
           userId_teamId: { userId: user.id, teamId: team.id },
         },
-        data: {
+        update: {
+          accepted: true,
+        },
+        create: {
+          userId: user.id,
+          teamId: team.id,
           accepted: true,
+          role: MembershipRole.MEMBER,
```

Comment:
Inviting by copy always adds a user as a **member**? Looks like we should support the role in it.

Reply:
Also, should we not create a membership for the parent team as well? I have seen in the codebase that when creating a membership we create a membership with parent team as well if it exists.
<img width="720" alt="Screenshot 2023-09-15 at 4 00 03 PM" src="https://github.com/calcom/cal.com/assets/1780212/e735e105-fc52-4524-b7df-805db90d3a4b">

Reply:
Noticedthat inviting to a team link is different and doesn't bring to signup page. So, this might be fine.